localstack-core/localstack/services/lambda_/event_source_mapping/esm_worker_factory.py

```diff
@@ -149,8 +157,23 @@ def get_esm_worker(self) -> EsmWorker:
                 processor=esm_processor,
             )
         else:
-            raise Exception(
-                f"Unsupported event source mapping source service {source_service}. Please upvote or create a feature request."
+            poller_holder = {"create_poller": None}
```

Comment:
It would be nice to add a type for that, so the hook signature can depend on it.

Reply:
A type for the `create_poller` key-value or a type for the entire payload being passed to the hook?

i.e a `PollerFromHook()` class that gets some internal value mutated in the hook

Reply:
Something like:

```python
class PollerHolder:
    def __init__(self):
        self.create_poller: Optional[Callable[..., Poller]] = None
```

That we pass in instead of the dict?